.github/workflows/docs.yaml

```diff
@@ -0,0 +1,40 @@
+name: Generate and publish docs
+
+on:
+  push:
+    branches: [ main ]
+
+env:
+  # Xcode 16.3 gets us Swift 6.1, required for docc merge
+  XCODE_VERSION: 16.3
+
+jobs:
+  build:
+    name: Generate API docs and publish to GitHub pages
+    # macos-15 is required for Xcode 16.3
+    runs-on: macos-15
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: jdx/mise-action@5083fe46898c414b2475087cc79da59e7da859e8
+      - uses: maxim-lobanov/setup-xcode@60606e260d2fc5762a71e64e74b2174e8ea3c8bd
+        with:
+          xcode-version: ${{ env.XCODE_VERSION }}
+
+      - name: Install dependencies
+        run: tuist install --path Samples
+
+      - name: Generate project
+        run: tuist generate --path Samples --no-open
+
+      - name: Generate Docs
+        run: Scripts/generate_docs.sh
+
+      - name: Deploy to GitHub Pages
+        uses: crazy-max/ghaction-github-pages@df5cc2bfa78282ded844b354faee141f06b41865
```

Comment:
is this the crazy-max we need or deserve? j/k, but how did you pick this flavor of page deployment? TCA repo uses [this one](https://github.com/JamesIves/github-pages-deploy-action), which 'wins' the GH star popularity contest (i don't know enough to know if they are comparable/equivalent though).

Reply:
This file is copied from [swift-modals](https://github.com/square/swift-modals/blob/main/.github/workflows/docs.yaml), which is in turn based on [Blueprint's](https://github.com/square/Blueprint/blob/main/.github/workflows/docs.yml). We've been using this action to publish Blueprint's jazzy docs for about 5 years. I couldn't tell you what the playing field looked like back then, but I know it's been reliable.

Reply:
When I was setting up `swift-modals` CI jobs I got some security warnings that suggested using SHAs instead of tags, so that's why they're versioned that way.